### PR TITLE
ogurec firmware

### PR DESCRIPTION
**Pull request by drhigsby**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ogurec/config.h [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define MANUFACTURER DrHigsby
+#define PRODUCT ogurec
+#define VENDOR_ID 0x0420
```
Comment (drashna) [inline keyboards/ogurec/config.h:23]: This Vendor ID is in use. 
https://www.the-sz.com/products/usbid/index.php?v=0x0420

Please select a different vendor ID.  Ideally, anything above 0x4000 isn't in use.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/ogurec/config.h (hunk 1 above)
Comment (drhigsby) [reply keyboards/ogurec/config.h:23]: I submitted this same vendor ID for dubba175 and it was accepted back on March 4th. 

https://github.com/qmk/qmk_firmware/blob/master/keyboards/dubba175/config.h

Should I change the vendor ID on dubba175 as well here for ogurec?

Comment (drhigsby) [reply keyboards/ogurec/config.h]: changed VID to 0x0007 (which produces no results at www.the-sz.com/products/usbid

File: keyboards/ogurec/keymaps/dack/keymap.c [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+#define BASE 0
+#define CLOSE 1
+#define FAR 2
+#define FUNKY 3
```
Comment (drashna) [inline keyboards/ogurec/keymaps/dack/keymap.c:21]:
```suggestion
enum custom_layers {
    _BASE,
    _CLOSE,
    _FAR,
    _FUNKY,
};
```

File: keyboards/ogurec/keymaps/dack/keymap.c [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+#define CLS_SPC LT(CLOSE, KC_SPC)
+#define FAR_F LT(FAR, KC_F)
+#define FNK_D LT(FUNKY, KC_D)
```
Comment (drashna) [inline keyboards/ogurec/keymaps/dack/keymap.c:24]:
```suggestion
#define CLS_SPC LT(_CLOSE, KC_SPC)
#define FAR_F LT(_FAR, KC_F)
#define FNK_D LT(_FUNKY, KC_D)
```

File: keyboards/ogurec/keymaps/dack/keymap.c [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+            return false;
+        }
+    }
+    return true;
+};
+
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [BASE] = LAYOUT_LEFT(
```
Comment (drashna) [inline keyboards/ogurec/keymaps/dack/keymap.c:59]:
```suggestion
    [_BASE] = LAYOUT_LEFT(
```

File: keyboards/ogurec/keymaps/dack/keymap.c [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+        KC_Q, KC_W, KC_E, KC_R, KC_T, KC_LBRC, KC_RBRC, KC_Y, KC_U, KC_I, KC_O, KC_P,
+        KC_A, KC_S, FNK_D, FAR_F, KC_G, KC_GRAVE, KC_SCOLON, KC_H, KC_J, KC_K, KC_L, KC_QUOT,
+        SFT_Z, CTL_X, WIN_C, ALT_V, KC_B, KC_BSPC, CLS_SPC, KC_N, ALT_M, WIN_COMM, CTL_DOT, SFT_ENT
+  ),
+
+    [CLOSE] = LAYOUT_LEFT(
```
Comment (drashna) [inline keyboards/ogurec/keymaps/dack/keymap.c:65]:
```suggestion
    [_CLOSE] = LAYOUT_LEFT(
```

File: keyboards/ogurec/keymaps/dack/keymap.c [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+        KC_1,     KC_2,    KC_3,    KC_4,    KC_5,    _______, _______, KC_6,    KC_7,       KC_8,     KC_9,    KC_0,
+        KC_EXLM,  KC_AT,   KC_HASH, KC_DLR,  KC_PERC, _______, _______, KC_CIRC, KC_AMPR,    KC_ASTR,  KC_LPRN, KC_RPRN,
+        KC_MINUS, KC_UNDS, KC_EQL,  KC_PLUS, _______, _______, vvv,     _______, KC_SCOLON,  KC_COLON, KC_SLSH, KC_QUES
+  ),
+
+    [FAR] = LAYOUT_LEFT(
```
Comment (drashna) [inline keyboards/ogurec/keymaps/dack/keymap.c:71]:
```suggestion
    [_FAR] = LAYOUT_LEFT(
```

File: keyboards/ogurec/keymaps/dack/keymap.c [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+        KC_ESC,  TYPEDON, KC_BSLS, RUS_TGL,_______,  _______,  _______,  KC_PGUP,   KC_HOME,    KC_UP,    KC_END,    KC_INS,
+        KC_CAPS, KC_TAB,  KC_PIPE, vvv,    LOCK,     _______,  _______,  KC_PGDN,   KC_LEFT,    KC_DOWN,  KC_RIGHT,  KC_DEL,
+        UNDO,    CUT,     COPY,    PASTE,  KC_LBRC,  KC_RBRC,  KC_LCBR,  KC_RCBR,   _______,    _______,  _______,   _______
+  ),
+    [FUNKY] = LAYOUT_LEFT(
```
Comment (drashna) [inline keyboards/ogurec/keymaps/dack/keymap.c:76]:
```suggestion
    [_FUNKY] = LAYOUT_LEFT(
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/ogurec/ogurec.h [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+/* This a shortcut to help you visually see your layout.
+ *
+ * The first section contains all of the arguments representing the physical
+ * layout of the board and position of the keys.
+ *
+ * The second converts the arguments into a two-dimensional array which
+ * represents the switch matrix.
+ */
+
+#define LAYOUT_RIGHT( \
```
Comment (drashna) [inline keyboards/ogurec/ogurec.h:29]:
```suggestion
#define LAYOUT_ortho_3x12( \
```

File: keyboards/ogurec/ogurec.h [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+    k100, k101, k102, k103, k104, k105, k106, k107, k108, k109, k110, k111,  \
+    k200, k201, k202, k203, k204, k205, k206, k207, k208, k209, k210, k211    \
+) \
+{ \
+    { k000, k001, k002, k003, k004, k005, k006, k007, k008, k009, k010, k011 }, \
+    { k100, k101, k102, k103, k104, k105, k106, k107, k108, k109, k110, k111 },  \
+    { k200, k201, k202, k203, k204, k205, k206, k207, k208, k209, k210, k211 }    \
+}
+
+#define LAYOUT_LEFT( \
```
Comment (drashna) [inline keyboards/ogurec/ogurec.h:40]:
```suggestion
#define LAYOUT_ortho_3x12_flipped( \
```

File: keyboards/ogurec/ogurec.h [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+    k211, k210, k209, k208, k207, k206, k205, k204, k203, k202, k201, k200    \
+) \
+{ \
+    { k000, k001, k002, k003, k004, k005, k006, k007, k008, k009, k010, k011 }, \
+    { k100, k101, k102, k103, k104, k105, k106, k107, k108, k109, k110, k111 },  \
+    { k200, k201, k202, k203, k204, k205, k206, k207, k208, k209, k210, k211 }    \
+}
+
+
+#define LAYOUT_RIGHT_2U( \
```
Comment (drashna) [inline keyboards/ogurec/ogurec.h:52]:
```suggestion
#define LAYOUT_ortho_3x12_1x2uC( \
```

File: keyboards/ogurec/ogurec.h [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+    k100, k101, k102, k103, k104, k105, k106, k107, k108, k109, k110, k111,  \
+    k200, k201, k202, k203, k204,    k205,    k207, k208, k209, k210, k211    \
+) \
+{ \
+    { k000, k001, k002, k003, k004, k005, k006, k007, k008, k009, k010, k011 }, \
+    { k100, k101, k102, k103, k104, k105, k106, k107, k108, k109, k110, k111 },  \
+    { k200, k201, k202, k203, k204, k205, KC_NO, k207, k208, k209, k210, k211 }   \
+}
+
+#define LAYOUT_LEFT_2U( \
```
Comment (drashna) [inline keyboards/ogurec/ogurec.h:63]:
```suggestion
#define LAYOUT_ortho_3x12_1x2uC_flipped( \
```

**Review by drashna — COMMENTED** [review]
Would need matching changes to keymaps too.

File: keyboards/ogurec/readme.md [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+
+![ogurec](https://imgur.com/gallery/OFF1Vyp)
+
+ogurec is a reversible 12u x 3u ortholinear keyboard that offers left-sided and right-sided PM placement
+
+* Keyboard Maintainer: [H. Bond](https://github.com/drhigsby)
+* Hardware Supported: ogurec (https://github.com/drhigsby/ogurec)
+* Hardware Availability: open source 3DP case available here https://github.com/drhigsby (case creator: https://github.com/dpalka15)
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/ogurec/readme.md:11]: Could you add instructions on how to enter the bootloader, to flash new firmware?

Comment (drhigsby) [reply keyboards/ogurec/readme.md:11]: I added instructions: https://github.com/qmk/qmk_firmware/pull/13242/commits/a607beb364466df11556c6a252ee52b087ba996a

**Review by drashna — APPROVED** [review]

File: keyboards/ogurec/rules.mk [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+LTO_ENABLE = no             # Use link time optimization
```
Comment (fauxpark) [inline keyboards/ogurec/rules.mk:23]:
```suggestion
```

File: keyboards/ogurec/readme.md [hunk 14]
```diff
@@ -0,0 +1,18 @@
+# ogurec
+
+![ogurec](https://imgur.com/gallery/OFF1Vyp)
```
Comment (fauxpark) [inline keyboards/ogurec/readme.md:3]:
```suggestion
![ogurec](https://i.imgur.com/OJ5UnXT.png)
```

File: keyboards/ogurec/readme.md [hunk 15]
```diff
@@ -0,0 +1,18 @@
...
+ogurec is a reversible 12u x 3u ortholinear keyboard that offers left-sided and right-sided PM placement
+
+* Keyboard Maintainer: [H. Bond](https://github.com/drhigsby)
+* Hardware Supported: ogurec (https://github.com/drhigsby/ogurec)
+* Hardware Availability: open source 3DP case available here https://github.com/drhigsby (case creator: https://github.com/dpalka15)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make ogurec:default_right
+    make ogurec:default_left
```
Comment (fauxpark) [inline keyboards/ogurec/readme.md:14]:
```suggestion
    make ogurec:default
```
This board needs a single `default` keymap.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/ogurec/readme.md (hunk 15 above)
Comment (drhigsby) [reply keyboards/ogurec/readme.md:14]: @fauxpark Hey homie, I don't have a keymap called "default." Can I have just: 

> make ogurec:default_right

Will that be okay?

Comment (fauxpark) [reply keyboards/ogurec/readme.md:14]: No, there are certain parts of QMK and its tooling that assume every keyboard has a `default` keymap, in particular https://qmk.fm/keyboards/.

Comment (drhigsby) [reply keyboards/ogurec/readme.md:14]: Okay, that sounds like a good idea. I will look into re-organizing it that way. Thank you!

Comment (drhigsby) [reply keyboards/ogurec/readme.md]: updates here: https://github.com/qmk/qmk_firmware/pull/13242/commits/f551b86424eb15f134f98661f8e1d791bb756470

Comment (fauxpark) [reply keyboards/ogurec/readme.md]: I didn't mean `left` and `right` keymaps, I meant splitting the keyboard folder itself into `ogurec/left` and `ogurec/right`, with their own set of layout macros and individual `default` keymaps.

Comment (drhigsby) [reply keyboards/ogurec/readme.md]: I made some significant changes to the repo to do just this, but am having a lot of issues with it now. Not sure what I did wrong.

**Review by fauxpark — CHANGES_REQUESTED** [review]
Please remove the .lnk file.

File: keyboards/ogurec/rules.mk [hunk 16]
```diff
@@ -0,0 +1,24 @@
...
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+DEFAULT_FOLDER = ogurec/left_pm
+LAYOUTS = ortho_3x12_grid
```
Comment (fauxpark) [inline keyboards/ogurec/rules.mk:24]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/ogurec/readme.md (hunk 15 above)
Comment (drhigsby) [reply keyboards/ogurec/readme.md]: okay, everything is now compiling correctly after further testing

File: keyboards/ogurec/rules.mk (hunk 16 above)
Comment (drhigsby) [reply keyboards/ogurec/rules.mk]: this has been updated

File: keyboards/ogurec/left_pm/readme.md [hunk 17]
```diff
@@ -0,0 +1,13 @@
+# ogurec keymap for left-sided PM placement
+
+Make example for this keyboard (after setting up your build environment):
+
+    make ogurec:default
```
Comment (fauxpark) [inline keyboards/ogurec/left_pm/readme.md:5]:
```suggestion
    make ogurec/left_pm:default
```

File: keyboards/ogurec/left_pm/readme.md [hunk 18]
```diff
@@ -0,0 +1,13 @@
...
+
+Flashing example for this keyboard:
+
+    make ogurec:default:flash
```
Comment (fauxpark) [inline keyboards/ogurec/left_pm/readme.md:9]:
```suggestion
    make ogurec/left_pm:default:flash
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/ogurec/left_pm/readme.md (hunk 17 above)
Comment (drhigsby) [reply keyboards/ogurec/left_pm/readme.md:5]: I actually specify DEFAULT_FOLDER = ogurec/left_pm in rules.mk. Should I commit this suggested change and remove this form rules.mk?

Comment (fauxpark) [reply keyboards/ogurec/left_pm/readme.md:5]: No, it's just that you have three readmes now, so the left and right specific ones should have the correct make command (even if left is default).

File: keyboards/ogurec/keymaps/dack/readme.md [hunk 19]
```diff
@@ -0,0 +1,3 @@
+# Dack's Ogurec Keymap
+
+This is Dack's personal ogurec keymap and it is for left-sided PM placement
```
Comment (fauxpark) [inline keyboards/ogurec/keymaps/dack/readme.md:3]: Just by the way, if you want to make this keymap unavailable to build for `right_pm` you can move it to another `keymaps/` folder under `left_pm`.

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks

**Pull request merged**